crates/bevy_color/src/color_gradient.rs

```diff
@@ -0,0 +1,104 @@
+use crate::Mix;
+use bevy_math::curve::{cores::EvenCoreError, Curve, Interval, SampleCurve};
+
+/// A curve whose samples are defined by a collection of colors.
+#[derive(Clone, Debug)]
+pub struct ColorCurve<T: Mix + Clone, I> {
+    curve: SampleCurve<T, I>,
+}
+
+impl<T, I> ColorCurve<T, I>
+where
+    T: Mix + Clone,
+{
```

Comment:
```suggestion
impl<T> ColorCurve<T>
where
    T: Mix + Clone,
{
```

Which lets us remove the `I` parameter here too.